### PR TITLE
[5.2] Fix \Illuminate\View\View::__construct docblock

### PR DESCRIPTION
**Pull request by lucasmichot**
Fix `\Illuminate\View\View::__construct` docblock

### CODE REVIEW

File: src/Illuminate/View/View.php [hunk 1]
```diff
@@ -58,7 +58,7 @@ class View implements ArrayAccess, ViewContract
      * @param  \Illuminate\View\Engines\EngineInterface  $engine
      * @param  string  $view
      * @param  string  $path
-     * @param  array   $data
+     * @param  mixed   $data
```
Comment (GrahamCampbell) [inline src/Illuminate/View/View.php]: Not correct.

Comment (GrahamCampbell) [inline src/Illuminate/View/View.php]: Please use the most specific doc possible.

Comment (GrahamCampbell) [inline src/Illuminate/View/View.php]: Arrayable|array

Comment (lucasmichot) [inline src/Illuminate/View/View.php]: What about ` (array) $data;`?

Comment (lucasmichot) [inline src/Illuminate/View/View.php]: `Arrayable|array` is not correct, I can pass an object too as argument.

Comment (GrahamCampbell) [inline src/Illuminate/View/View.php]: object|array then

Comment (lucasmichot) [inline src/Illuminate/View/View.php]: `\Illuminate\Contracts\Support\Arrayable|object|array` => `mixed`
As for https://github.com/laravel/framework/blob/5.2/src/Illuminate/Support/Collection.php#L1135-L1156

Comment (GrahamCampbell) [inline src/Illuminate/View/View.php]: Illuminate\Contracts\Support\Arrayable is a subset of object

Comment (GrahamCampbell) [inline src/Illuminate/View/View.php]: > https://github.com/laravel/framework/blob/5.2/src/Illuminate/Support/Collection.php#L1135-L1156

That link is not idea. We always prefer the most useful phpdoc. Saying mixed is a bit of a waste if a phpdoc.

**Pull request merged**